akka-projection-grpc/src/test/scala/akka/projection/grpc/internal/ProtoAnySerializationSpec.scala

```diff
@@ -72,35 +90,142 @@ class ProtoAnySerializationSpec extends ScalaTestWithActorTestKit with AnyWordSp
       val typeUrl = "type.my.io/custom"
       val event =
         ScalaPbAny(typeUrl, ByteString.copyFrom(value, StandardCharsets.UTF_8))
-      val pbAny = protoAnySerialization.encode(event)
+      val pbAny = serializationJava.serialize(event)
       pbAny.typeUrl shouldBe typeUrl
       val deserializedEvent =
-        protoAnySerialization.decode(pbAny).asInstanceOf[PbAny]
+        serializationJava.deserialize(pbAny).asInstanceOf[PbAny]
       deserializedEvent.getTypeUrl shouldBe typeUrl
       deserializedEvent.getValue.toString(StandardCharsets.UTF_8) shouldBe value
     }
 
     "encode and decode with Akka serialization with string manifest" in {
       val event = Address("akka", system.name, "localhost", 2552)
-      val pbAny = protoAnySerialization.encode(event)
+      val pbAny = serializationJava.serialize(event)
       val serializer = akkaSerialization.findSerializerFor(event)
       // no manifest for String serializer
       pbAny.typeUrl shouldBe s"ser.akka.io/${serializer.identifier}:${Serializers
         .manifestFor(serializer, event)}"
 
-      val deserializedEvent = protoAnySerialization.decode(pbAny)
+      val deserializedEvent = serializationJava.deserialize(pbAny)
       deserializedEvent shouldBe event
     }
 
     "encode and decode with Akka serialization without string manifest" in {
       val event = "e1"
-      val pbAny = protoAnySerialization.encode(event)
+      val pbAny = serializationJava.serialize(event)
       val serializer = akkaSerialization.findSerializerFor(event)
       // no manifest for String serializer
       pbAny.typeUrl shouldBe s"ser.akka.io/${serializer.identifier}"
 
-      val deserializedEvent = protoAnySerialization.decode(pbAny)
+      val deserializedEvent = serializationJava.deserialize(pbAny)
       deserializedEvent shouldBe event
     }
+
+    "support se/deserializing java protobufs" in {
+      val any = serializationJava.encode(addLineItem)
+      any.typeUrl should ===("type.googleapis.com/" + AddLineItem.scalaDescriptor.fullName)
+      serializationJava.decodePossiblyPrimitive(any) should ===(addLineItem)
+    }
+
+    def testPrimitive[T](name: String, value: T, defaultValue: T) = {
+      val any = serializationJava.encode(value)
+      any.typeUrl should ===(ProtoAnySerialization.PrimitivePrefix + name)
+      serializationJava.decodePossiblyPrimitive(any) should ===(value)
+
+      val defaultAny = serializationJava.encode(defaultValue)
+      defaultAny.typeUrl should ===(ProtoAnySerialization.PrimitivePrefix + name)
+      defaultAny.value.size() shouldBe 0
+      serializationJava.decodePossiblyPrimitive(defaultAny) should ===(defaultValue)
+    }
+
+    "support se/deserializing strings" in testPrimitive("string", "foo", "")
```

Comment:
These special cases for StringValue/BytesValue, they are mostly in the Kalix SDK for Kalix-and-Kalix-SDK-design-reasons, for example in nested usage in the replicated APIs, I think they should rather work like any other protobuf message types here, and that user events will seldom be primitives anyway. (or am I missing some use case?)

Reply:
IMO drop both because so unlikely that it will be used often